### DISCUSSION TITLE
php bakery does end with error Code

### DISCUSSION THREAD

**Discussion by X-Nightmare-X** [userfrosting/UserFrosting#1279] [Q&A & Troubleshooting]
I wanted to install UserFrosting like mentioned here:
https://learn.userfrosting.com/installation/environment/native/install

I used php composer.phar.

I am running into this failure.

```
UserFrosting Environnement Information
======================================
In VersionParser.php line 191:                 
  Invalid version string "" 
bake
Script php bakery bake handling the post-create-project-cmd event returned with error code 1
```

Now i dont know what to do.
If i run "php bakery bake"
It says this:
```
sh: Zeile 1: node: Kommando nicht gefunden.
sh: Zeile 1: npm: Kommando nicht gefunden.

 _   _              ______             _   _
| | | |             |  ___|           | | (_)
| | | |___  ___ _ __| |_ _ __ ___  ___| |_ _ _ __   __ _
| | | / __|/ _ \ '__|  _| '__/ _ \/ __| __| | '_ \ / _` |
| |_| \__ \  __/ |  | | | | | (_) \__ \ |_| | | | | (_| |
 \___/|___/\___|_|  \_| |_|  \___/|___/\__|_|_| |_|\__, |
                                                    __/ |
                                                   |___/

UserFrosting's Database Setup Wizard
====================================

 ! [NOTE] Database already setup. Use the `php bakery setup:db --force` command 
 !        to run db setup again.                                                


UserFrosting's Mail Setup Wizard
================================

 ! [NOTE] Mail appears to be already setup in .env file. Use the `php bakery    
 !        setup:mail --force` command to run setup again.                       


UserFrosting Environnement Information
======================================


In VersionParser.php line 191:
                             
  Invalid version string ""  
                             

bake
```

Please help, that i can install the login and managing page and then add my own subsites.

**Comment by lcharette** [comment]
You don't have Node and NPM installed : https://learn.userfrosting.com/installation/environment/native/requirements#node

**Comment by X-Nightmare-X** [comment]
Jeah not from command line but i activated it in the web env.
I can exec commands through the web interface.

But i read, that i can also install it on another system and then upload it to the webhost and run it there without it?!

I dont know why the cli doenst support the commands. I think for security reasons.
I have composer also but in CLI i have to use composer.phar..

I think i will ask my webhoster if they can also activate it in the CLI

But what do i have to exec then with npm/node?

Or would it be better to run it via the script in CLI? Then on another host und reupload?

**Comment by X-Nightmare-X** [comment]
Ok,  now i tested it on my root Server. Now I am getting the following after entering my SQL Server data

` Could not connect to the database 'XYZ@123.123.123.123/XYZ':                    
         Exception: could not find driver                                                                               
                                                                                                                        
         Please check your database configuration and/or google the exception shown above and run the command again.    
                                                                                                                        

Script php bakery bake handling the post-create-project-cmd event returned with error code 1`

I installed php-mysql because i thought it would be that. 
But didnt work also.

Do you know what is missing?
Which package do i have to install?

**Comment by X-Nightmare-X** [comment]
I ran the command like in the documention said. Now my question ist how to get it to run on my Website.
The website after uploading all files says:

```
We're sorry, but something went wrong.
The issue has been logged for investigation. Please try again later.
Web application could not be started by the Phusion Passenger(R) application server.
```

**Reply by lcharette** [reply]
You'll need to investigate the log to know more I'm afraid.
